core/domain/feedback_jobs_continuous_test.py

```diff
@@ -60,18 +60,85 @@ def _get_swap_context(self):
             jobs_registry, 'ALL_CONTINUOUS_COMPUTATION_MANAGERS',
             self.ALL_CC_MANAGERS_FOR_TESTS)
 
-    def _run_job_and_check_results(self, exp_id,
-                                   expected_thread_analytics_dict):
+    def _run_job(self):
         self.process_and_flush_pending_tasks()
         ModifiedFeedbackAnalyticsAggregator.start_computation()
         self.assertEqual(self.count_jobs_in_taskqueue(), 1)
         self.process_and_flush_pending_tasks()
 
+    def _run_job_and_check_results(self, exp_id,
+                                   expected_thread_analytics_dict):
+        self._run_job()
         self.assertEqual(
             ModifiedFeedbackAnalyticsAggregator.get_thread_analytics(
                 exp_id).to_dict(),
             expected_thread_analytics_dict)
 
+    def test_no_threads_job_not_run(self):
+        # Create a new exploration, but don't create any threads and don't even
+        # start the MR job.
+        with self._get_swap_context():
+            exp_id = 'eid'
+            self.save_new_valid_exploration(exp_id, 'owner')
+            self.assertEqual(
+                ModifiedFeedbackAnalyticsAggregator.get_thread_analytics(
+                    exp_id).to_dict(), {
+                        'num_open_threads': 0,
+                        'num_total_threads': 0,
+                    })
+
+    def test_get_thread_analytics_multi(self):
+        with self._get_swap_context():
+            exp_id_1 = 'eid1'
+            exp_id_2 = 'eid2'
+            thread_id = 'tid'
+            self.save_new_valid_exploration(exp_id_1, 'owner')
+            self.save_new_valid_exploration(exp_id_2, 'owner')
+
+            initial_feedback_threads = (
+                ModifiedFeedbackAnalyticsAggregator.get_thread_analytics_multi(
+                    [exp_id_1, exp_id_2]))
+            self.assertEqual(len(initial_feedback_threads), 2)
+            self.assertEqual(initial_feedback_threads[0].to_dict(), {
+                'num_open_threads': 0,
+                'num_total_threads': 0,
+            })
+            self.assertEqual(initial_feedback_threads[1].to_dict(), {
+                'num_open_threads': 0,
+                'num_total_threads': 0,
+            })
+
+            thread = feedback_models.FeedbackThreadModel.create(
+                exp_id_1, thread_id)
+            thread.exploration_id = exp_id_1
+            thread.put()
+            feedback_threads = (
+                ModifiedFeedbackAnalyticsAggregator.get_thread_analytics_multi(
+                    [exp_id_1, exp_id_2]))
+            self.assertEqual(len(feedback_threads), 2)
+            self.assertEqual(feedback_threads[0].to_dict(), {
+                'num_open_threads': 0,
```

Comment:
(Look in core.domain.feedback_services. The realtime models are updated using events -- look for the event_services imports.)